src/components/app/routes/data/utils.js

```diff
@@ -278,23 +279,24 @@ export async function ensureActiveEnterpriseCustomerUser({
   // Otherwise, try to find the enterprise customer for the given slug and, if found, update it
   // as the active enterprise customer for the learner.
   const foundEnterpriseCustomerUserForSlug = allLinkedEnterpriseCustomerUsers.find(
-    enterpriseCustomerUser => enterpriseCustomerUser.enterpriseCustomer?.slug === enterpriseSlug,
```

Comment:
[inform] This was previously incorrectly returning an enterprise customer when linked to enterprise customer(s) where `enable_learner_portal: false` and there is no `enterpriseSlug` specified in the page route URL (`undefined` === `undefined`.

The fix ensures we only try to match a customer slug if the customer actually exists, and similarly only attempt to update the active enterprise customer when a slug is the page route URL as a secondary gating approach.